src/libraries/Microsoft.Extensions.Hosting/docs/HostShutdown.md

```diff
@@ -0,0 +1,68 @@
+# Host Shutdown
+
+A Hosted Service process can be stopped in the following ways:
+
+1. If someone doesn't call `Run` or `WaitForShutdown` and the app exits normally with "Main" completing
+2. A crash occurs
+3. The app is forcefully shut down, e.g. SIGKILL (i.e. CTRL+Z)
+4. When ConsoleLifetime is used it listens for the following signals and attempts to stop the host gracefully
+    1. SIGINT (i.e. CTRL+C)
+    2. SIGQUIT (i.e. CTRL+BREAK on Windows, CTRL+`\` on Unix)
+    3. SIGTERM (sent by other apps, e.g. `docker stop`)
+5. The app calls `Environment.Exit(code)`
+
+Scenarios (1), (2), and (3) aren't handled directly by the Hosting code. The owner of the process needs to deal with
+them the same as any application.
+
+Scenarios (4) and (5) are handled specially by the built-in Hosting logic. Specifically by the ConsoleLifetime
+class. ConsoleLifetime tries to handle the "shutdown" signals listed in (4) and allow for a graceful exit to the
+application.
+
+Before .NET 6, there wasn't a way for .NET code to gracefully handle SIGTERM. To work around this limitation,
+ConsoleLifetime would subscribe to `AppDomain.CurrentDomain.ProcessExit`. When `ProcessExit` was raised,
+ConsoleLifetime would signal the Host to stop, and block the `ProcessExit` thread, waiting for the Host to stop.
+This would allow for the clean up code in the application to run - for example, `IHostedService.StopAsync` and code after
+`Host.Run` in the Main method.
+
+This caused other issues because SIGTERM wasn't the only way `ProcessExit` was raised. It is also raised by code
+in the application calling `Environment.Exit`. `Environment.Exit` isn't a graceful way of shutting down a process.
```

Comment:
```suggestion
in the application calling `Environment.Exit`. `Environment.Exit` isn't a graceful way of shutting down a process in the `Microsoft.Extensions.Hosting` app model.
```

Reply:
Is it a graceful way of shutting down a process in any app model?

Reply:
`Environment.Exit` is a graceful program termination from the runtime point of view.

App models typically want to wrap exit with their own logic. I agree that it is probably not considered the right way to exit the process in any app model, except the "no app model" simple apps.

Reply:
I’ll fix this up in a follow up PR tomorrow so I don’t need to reset CI.